ui/components/multichain/network-list-menu/add-rpc-url-modal/add-rpc-url-modal.tsx

```diff
@@ -1,42 +1,109 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import {
   Box,
   ButtonPrimary,
   ButtonPrimarySize,
   FormTextField,
+  FormTextFieldSize,
+  HelpText,
+  HelpTextSeverity,
 } from '../../../component-library';
 import {
+  BackgroundColor,
   BlockSize,
+  BorderRadius,
   Display,
+  FlexDirection,
+  JustifyContent,
   TextVariant,
 } from '../../../../helpers/constants/design-system';
 import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { isWebUrl } from '../../../../../app/scripts/lib/util';
 
-const AddRpcUrlModal = () => {
+const AddRpcUrlModal = ({
+  onAdded,
+}: {
+  onAdded: (url: string, name?: string) => void;
+}) => {
   const t = useI18nContext();
 
+  const [url, setUrl] = useState<string>();
+  const [error, setError] = useState<string>();
+  const nameRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (url && url?.length > 0 && !isWebUrl(url)) {
```

Comment:
is `url?.length > 0` necessary?

Reply:
probably not, the truthy check should handle it

Reply:
https://github.com/MetaMask/metamask-extension/pull/26433/commits/6dc153bc16641a1a1e65a2aa6e899cbfdc87d948